src/main/java/dev/the_fireplace/unforgivingvoid/usecase/SpawnPositionLocator.java

```diff
@@ -122,23 +118,27 @@ public BlockPos findSpawnPosition(EntityType<?> entityType, ServerWorld targetWo
                 return targetFocalPosition;
             }
 
-            int targetX = applyHorizontalOffset(rand, targetFocalPosition.getX());
-            int targetZ = applyHorizontalOffset(rand, targetFocalPosition.getZ());
-            int targetY = targetWorld.getTopY(Heightmap.Type.WORLD_SURFACE, targetX, targetZ);
-            BlockPos attemptPos = new BlockPos(targetX, targetY, targetZ);
-
-            spawnVec = findSafePlatform(entityType, targetWorld, attemptPos);
+            spawnVec = findSafePlatform(entityType, targetWorld, targetFocalPosition);
         }
 
         return new BlockPos(spawnVec.get());
     }
 
-    private Optional<Vec3d> findSafePlatform(EntityType<?> entityType, ServerWorld targetWorld, BlockPos blockPos) {
-        return safePosition.findBy(entityType, targetWorld, blockPos);
+    private Optional<Vec3d> findSafePlatform(EntityType<?> entityType, ServerWorld targetWorld, BlockPos targetFocalPosition) {
+        Random rand = targetWorld.getRandom();
+
+        int targetX = applyHorizontalOffset(rand, targetFocalPosition.getX());
+        int targetZ = applyHorizontalOffset(rand, targetFocalPosition.getZ());
+        int targetY = targetWorld.getTopY(Heightmap.Type.WORLD_SURFACE, targetX, targetZ);
```

Comment:
In `findSimilarPosition` I'd still like them to be able to spawn in caves. Setting targetY here replaces that and makes it identical to `findSurfacePosition`

Reply:
Ah, my bad. I was refactoring and didn't notice that.

Let me update that rn